corehq/apps/domain/views/accounting.py

```diff
@@ -1947,3 +1950,74 @@ def pause_subscription(request, domain):
     return HttpResponseRedirect(
         reverse(DomainSubscriptionView.urlname, args=[domain])
     )
+
+
+@require_POST
+@login_and_domain_required
+@require_permission(HqPermissions.edit_billing)
+def enable_subscription_auto_renew(request, domain):
+    current_subscription = Subscription.get_active_subscription_by_domain(domain)
+    if not current_subscription.user_can_change_subscription(request.user):
+        messages.error(
+            request, _(
+                "You do not have permission to modify the subscription for this customer-level account. "
+                "Please reach out to the %s enterprise admin for help."
+            ) % current_subscription.account.name
+        )
+        return HttpResponseRedirect(
+            reverse(DomainSubscriptionView.urlname, args=[domain])
+        )
+
+    next_subscription = current_subscription.next_subscription
+    if next_subscription is not None:
+        # can't enable auto renewal because we don't know what to do with the next subscription
+        # this is blocked via UI but validate here as well
+        messages.error(
+            request, _("Auto renewal cannot be enabled for the current subscription.")
```

Comment:
Should the user know why they can't enable auto renewal here? Unsure if the user can do anything to fix it

Reply:
They can't, because there's no user action that can directly change or delete a future subscription. Because this won't be accessible from the UI, they would have had to try to do something they shouldn't, like POST to the url manually, to try to make this happen. I'm not really worried.